csaf-rs/src/wasm.rs

```diff
@@ -0,0 +1,104 @@
+//! WASM bindings for CSAF validation
+//!
+//! This module provides WebAssembly bindings for validating CSAF documents in the browser.
+
+use crate::csaf::csaf2_0::loader::load_document_from_str as load_document_from_str_2_0;
+use crate::csaf::csaf2_1::loader::load_document_from_str as load_document_from_str_2_1;
+use crate::csaf::validation::{ValidationPreset, ValidationResult, validate_by_preset};
+use wasm_bindgen::prelude::*;
+
+/// Initialize panic hook for better error messages in the browser console
+#[wasm_bindgen(start)]
+pub fn init() {
+    console_error_panic_hook::set_once();
+}
+
+/// Validate a CSAF document from JSON string
+///
+/// This function auto-detects the CSAF version from the document and validates it
+/// according to the specified preset.
+///
+/// # Arguments
+///
+/// * `json_str` - The CSAF document as a JSON string
+/// * `preset` - The validation preset to use ("basic", "extended", or "full")
+///
+/// # Returns
+///
+/// A `ValidationResult` containing the validation outcome and any errors
+#[wasm_bindgen(js_name = validateCsaf)]
+pub fn validate_csaf(json_str: &str, preset_str: &str) -> Result<ValidationResult, JsValue> {
+    // Parse the preset
+    let preset = preset_str.parse::<ValidationPreset>().map_err(|_| {
+        JsValue::from_str(&format!(
+            "Invalid preset: {}. Use 'basic', 'extended', or 'full'",
```

Comment:
I might be missing something, but there are also the presets `mandatory`, `optional`/`recommended`, `informative`...

Reply:
Yes, that error message is not good :)

Reply:
And CSAF 2.1 adds a few more...